tests/database/test_database_manager.py

```diff
@@ -29,60 +31,44 @@ class ORMMappedTable(Base):
     value = Column(String)
 
 
-class TestDatabaseManager(TestCase):
-    """
-    Test the DatabaseManager module.
-    """
-
-    TEST_VALUE = "Hello World"
-    TEST_IDENT = 10
-
-    def setUp(self) -> None:
-        """Set up the database and ensure it is empty."""
-        self.db_path = str(Path(__file__).parent.joinpath("macaron.db"))
-        self.db_man = DatabaseManager(self.db_path, base=Base)
-        con = sqlite3.connect(self.db_path)
-        with con:
-            con.execute("drop table if exists test_table;")
-            con.execute("drop table if exists new_test_table;")
-            con.execute("drop table if exists _test_table;")
-            con.execute("drop table if exists _test_orm_table;")
-            con.execute("drop view if exists test_orm_table;")
-            con.commit()
-
-    @staticmethod
-    def _assert_query_result(db_path: str, query: str, expect: list[tuple]) -> None:
-        con = sqlite3.connect(db_path)
-        with con:
-            cursor = con.execute(query)
-            rows = cursor.fetchall()
-            assert str(rows) == str(expect)
-
-    def test_insert(self) -> None:
-        """Insert to database using core api."""
-        tbl = Table("test_table", Base.metadata, Column("test", String))
-        self.db_man.create_tables()
-        self.db_man.insert(tbl, {"test": self.TEST_VALUE})
-        self._assert_query_result(self.db_path, "select * from test_table;", [(self.TEST_VALUE,)])
-
-    def test_context_manager_orm(self) -> None:
-        """Connect to database using context manager and create tables, with concurrent connections."""
-        with DatabaseManager(self.db_path, base=Base) as db_man:
-            table = ORMMappedTable(id=10, value=self.TEST_VALUE)
-            db_man.create_tables()
-            db_man.add_and_commit(table)
-            db_man.session.flush()
-
-        self._assert_query_result(self.db_path, "select * from _test_orm_table;", [(self.TEST_IDENT, self.TEST_VALUE)])
-        self._assert_query_result(self.db_path, "select * from test_orm_table;", [(self.TEST_IDENT, self.TEST_VALUE)])
-
-        # Interacting with db manager after ended session should not crash
-        db_man.add_and_commit(ORMMappedTable(id=100, value=self.TEST_VALUE))
-        tbl = Table("new_test_table", Base.metadata, Column("test", String))
-        db_man.create_tables()
-        db_man.insert(tbl, {"test": self.TEST_VALUE})
-        self._assert_query_result(self.db_path, "select * from new_test_table;", [(self.TEST_VALUE,)])
-
-    def tearDown(self) -> None:
-        """Terminate the database connection."""
-        self.db_man.terminate()
+DB_PATH = str(Path(__file__).parent.joinpath("macaron.db"))
+
+
+@pytest.fixture()
+def db_man() -> Iterable:
+    """Set up the database and ensure it is empty."""
+    db_manager = DatabaseManager(DB_PATH, base=Base)
+    con = sqlite3.connect(DB_PATH)
+    with con:
+        con.execute("drop table if exists test_table;")
```

Comment:
I wonder where are all of the following table that we try to drop come from? (e.g `test_table`, `new_test_table`, etc.) Are they created in any other tests?

Reply:
They were actually leftovers. I've removed them, see commit [6fb65cc](https://github.com/oracle/macaron/pull/305/commits/6fb65cc2c11569016090ceb8783f99564370da83)